designs/2019-core-options/README.md

```diff
@@ -0,0 +1,139 @@
+- Start Date: 2019-05-12
+- RFC PR: https://github.com/eslint/rfcs/pull/22
+- Authors: Toru Nagashima ([@mysticatea](https://github.com/mysticatea))
+
+# Configuring core options in Config Files
+
+## Summary
+
+This RFC adds several properties into config files to configure core options. People can use config files (including shareable configs!) instead of CLI options in order to configure some linter behavior.
+
+## Motivation
+
+We cannot configure some linter's behavior with config files, especially, shareable configs. It's convenient if we can configure those in shareable configs.
+
+## Detailed Design
+
+This RFC adds four properties to config files.
+
+```js
+// .eslintrc.js
+module.exports = {
+    noInlineConfig: false,              // Corresponds to --no-inline-config
+    reportUnusedDisableDirectives: false,    // Corresponds to --report-unused-disable-directives
+    verifyOnRecoverableParsingErrors: false, // Corresponds to --verify-on-recoverable-parsing-errors
+    ignorePatterns: [],                      // Corresponds to --ignore-pattern
+
+    overrides: [
+        {
+            files: ["*.ts"],
+            noInlineConfig: false,
+            reportUnusedDisableDirectives: false,
+            verifyOnRecoverableParsingErrors: false,
+            // ignorePatterns: [], // Forbid this to avoid confusion with 'excludedFiles' property.
+        },
+    ],
+}
+```
+
+### § noInlineConfig
+
+That value can be a boolean value. Default is `false`.
+
+If `true` then it disables inline directive comments such as `/*eslint-disable*/`.
+
+If `noInlineConfig` is `true`, `--no-inline-config` was not given, and there are one or more directive comments, then ESLint reports each directive comment as a warning message (`severify=1`). For example, `"'eslint-disable' comment was ignored because your config file has 'noInlineConfig' setting."`. Therefore, end-users can know why directive comments didn't work.
+
```

Comment:
I don't think it's a good practice that `noInlineConfig` and `--no-inline-config` have a different behaviour. (presently --no-inline-config just ignore inline configs, and **not** report warnings. )

Reply:
The reason for this is explicitness. The CLI option is what the end-user specified explicitly. However, the new option in config files is not so. If a shareable config specified the option, the end-user may be confused. If the end-user wanted to hide the message, they can use `--no-inline-config` CLI option explicitly.

Reply:
I agreed the `noInlineConfig`(in this PR) is the correct meaning as its word. `--no-inline-conifg` is more like `--ignore-inline-config`, I image we could rename the option in future versions. but it's out of this PR. :)